tonic/models/vic/grid_params.py

```diff
@@ -55,88 +93,109 @@ def __init__(self, nlayers=3, snow_bands=5):
-                                                               + 13)),
-                                       ('Wpwp_FRACT', np.arange(10 * nlayers
-                                                                + 13,
-                                                                11 * nlayers
-                                                                + 13)),
-                                       ('rough', np.array([11 * nlayers
-                                                           + 13])),
-                                       ('snow_rough', np.array([11 * nlayers
-                                                                + 14])),
-                                       ('annual_prec', np.array([11 * nlayers
-                                                                + 15])),
-                                       ('resid_moist', np.arange(11 * nlayers
-                                                                 + 16,
-                                                                 12 * nlayers
-                                                                 + 16)),
-                                       ('fs_active', np.array([12 * nlayers
-                                                               + 16])),
-                                       ])
-
-        self.snow_param = OrderedDict([('cellnum', np.array([0])),
-                                       ('AreaFract', np.arange(1,
-                                                               snow_bands
-                                                               + 1)),
-                                       ('elevation', np.arange(snow_bands + 1,
-                                                               2 * snow_bands
-                                                               + 1)),
-                                       ('Pfactor', np.arange(2 * snow_bands
-                                                             + 1,
-                                                             3 * snow_bands
-                                                             + 1))])
-
+                                       ('c', np.array([8]))])
+
+        i = 9
+        for var in ['expt', 'Ksat', 'phi_s', 'init_moist']:
+            self.soil_param[var] = np.arange(i, i + nlayers)
+            i += nlayers
+
+        self.soil_param['elev'] = np.array([i])
+        i += 1
+        self.soil_param['depth'] = np.arange(i, i + nlayers)
+        i += nlayers
+        self.soil_param['avg_T'] = np.array([i])
+        i += 1
+        self.soil_param['dp'] = np.array([i])
+        i += 1
+
+        varnames = ['bubble', 'quartz', 'bulk_density', 'soil_density']
+        if organic_fract:
+            varnames.append(['organic', 'bulk_dens_org', 'soil_dens_org'])
+        for var in varnames:
+            self.soil_param[var] = np.arange(i, i + nlayers)
+            i += nlayers
+
+        self.soil_param['off_gmt'] = np.array([i])
+        i += 1
+
```

Comment:
OK, I'm making this more dynamic, although it might still be somewhat clunky.  Can we improve the code on a separate PR?